### DISCUSSION TITLE
ersatztv - Add HW transcoding to script

### DISCUSSION THREAD

**Discussion by Go2Engle** [tteck/Proxmox#3749] [Ideas]
The way the helper script passes through the HW transcoding for the jellyfin script works amazing. Could you add that to the ersatztv script?

**Comment by MickLesk** [comment]
Need to check by tteck, but i think its done ;)

https://github.com/tteck/Proxmox/pull/3750

**Reply by Go2Engle** [reply]
woo awesome! if it gets merged ill test it out!

**Comment by Go2Engle** [comment]
@MickLesk I seen your PR was closed. I just wanted to let you know in your sed I seen you were trying to add user ersatztv. ersatztv seems to run as root and does not have its own user like jellyfin does.

**Comment by Go2Engle** [comment]
@MickLesk If you plan on opening another PR I ran through manually and you can just skip the sed portion of the script. Transcoding is working with just the below. 
```shell
msg_info "Setting Up Hardware Acceleration"
$STD apt-get -y install {va-driver-all,ocl-icd-libopencl1,intel-opencl-icd,vainfo,intel-gpu-tools}
if [[ "$CTTYPE" == "0" ]]; then
  chgrp video /dev/dri
  chmod 755 /dev/dri
  chmod 660 /dev/dri/*
  $STD adduser $(id -u -n) video
  $STD adduser $(id -u -n) render
fi
msg_ok "Set Up Hardware Acceleration"

```

**Comment by Go2Engle** [comment]
@MickLesk Woops i spoke too soon! one other thing I needed to do to get some of my media to play.

I had to modify `/etc/apt/sources.list` from:
```shell
deb http://deb.debian.org/debian bookworm main contrib

deb http://deb.debian.org/debian bookworm-updates main contrib

deb http://security.debian.org bookworm-security main contrib
```
too:
```shell
deb http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware

deb http://deb.debian.org/debian bookworm-updates main contrib non-free

deb http://security.debian.org bookworm-security main contrib non-free
```
Then I installed `intel-media-va-driver-non-free` via
```shell
apt-get install -y intel-media-va-driver-non-free
```

Now everything seems to be transcoding!

**Reply by MickLesk** [reply]
I Take an Look tomorrow :-)

**Reply by Go2Engle** [reply]
to make it a breeze for ya lol

```shell
# Backup the original /etc/apt/sources.list
cp /etc/apt/sources.list /etc/apt/sources.list.bak

# Update /etc/apt/sources.list with new contents
cat <<EOF > /etc/apt/sources.list
deb http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware
deb http://deb.debian.org/debian bookworm-updates main contrib non-free
deb http://security.debian.org bookworm-security main contrib non-free
EOF

# Update package lists
apt-get update

# Install the intel-media-va-driver-non-free package
apt-get install -y intel-media-va-driver-non-free

# Output success message
echo "Sources list updated and intel-media-va-driver-non-free installed successfully."
```

**Reply by tteck** [reply]
No iGPU device has been passed to the LXC. HW transcoding is not possible.

**Reply by Go2Engle** [reply]
@tteck  I can confirm it is working.
I setup a fresh Ersatztv and attached devices exactly as jellyfin script does
![image](https://github.com/user-attachments/assets/e3b62da1-7ae7-426d-bb98-3fa1ef66eb39)
Then in the CT I ran the below
```shell
$STD apt-get -y install {va-driver-all,ocl-icd-libopencl1,intel-opencl-icd,vainfo,intel-gpu-tools}
chgrp video /dev/dri
chmod 755 /dev/dri
chmod 660 /dev/dri/*
$STD adduser $(id -u -n) video
$STD adduser $(id -u -n) render

cat <<EOF > /etc/apt/sources.list
deb http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware
deb http://deb.debian.org/debian bookworm-updates main contrib non-free
deb http://security.debian.org bookworm-security main contrib non-free
EOF

apt-get update

apt-get install -y intel-media-va-driver-non-free
```
Then I rebooted for good measure. 
In ErsatzTV I modified the FFmpeg Profile to the below and everything is working and transcoding via the iGPU verified using `intel_gpu_top` on the host.
![image](https://github.com/user-attachments/assets/03832523-67eb-4940-b271-08b8aabc8055)

**Reply by Go2Engle** [reply]
here is the output of `intel_gpu_top` while playing a channel in ErsatzTV
![image](https://github.com/user-attachments/assets/2abc3ab2-f310-44fa-b608-54f039463b74)

**Reply by tteck** [reply]
Aww, you manually passed the device.

**Reply by Go2Engle** [reply]
Yeah I just mimicked what the jellyfin script does just manually.

**Reply by Go2Engle** [reply]
I had seen that @MickLesk already took care of the `build.func` changes 
https://github.com/MickLesk/Proxmox/commit/817ab5dd8aefff21ff31178267409064d045b92d

**Reply by Go2Engle** [reply]
@MickLesk opened a PR on your fork https://github.com/MickLesk/Proxmox/pull/2 that should be the only changes needed for you to reopen your PR.

**Comment by tteck** [comment]
Hardware Acceleration has been added.

**Reply by Go2Engle** [reply]
Woo awesome! I see you didnt add the `intel-media-va-driver-non-free` though. I needed that for it to fully function and I'd imagine others may need it as well.

**Reply by tteck** [reply]
If Jellyfin works for you, you shouldn't need non-free.

**Reply by Go2Engle** [reply]
it is strange the same media transcoding in jellyfin does work fine. but It was not working in ErsatzTV. The Channel when loaded had errors saying it could not transcode. I dont have screenshot of the error. As soon as I installed the non-free the media was then able to be transcoded with no issues.

**Reply by Go2Engle** [reply]
gonna spin up a new instance with your changes and see if i can reproduce.

**Reply by Go2Engle** [reply]
Yep here is the error with the fresh instance.
![image](https://github.com/user-attachments/assets/282a5525-f52e-4b37-9601-27f8854c06f9)

**Reply by Go2Engle** [reply]
and after running the below it now works. Pixelated screenshot below xD
```shell
# Backup the original /etc/apt/sources.list
cp /etc/apt/sources.list /etc/apt/sources.list.bak

# Update /etc/apt/sources.list with new contents
cat <<EOF > /etc/apt/sources.list
deb http://deb.debian.org/debian bookworm main contrib non-free non-free-firmware
deb http://deb.debian.org/debian bookworm-updates main contrib non-free
deb http://security.debian.org bookworm-security main contrib non-free
EOF

# Update package lists
apt-get update

$STD apt-get -y install {intel-media-va-driver-non-free,va-driver-all,ocl-icd-libopencl1,intel-opencl-icd,vainfo,intel-gpu-tools}
```

![image](https://github.com/user-attachments/assets/45c35a44-ae2b-47be-91c1-90118f974afb)
